UnityProject/Assets/Scripts/Construction/ConveyorBelt.cs

```diff
@@ -0,0 +1,350 @@
+	private ConveyorStatus LastStatus;
+
+	[SyncVar(hook = nameof(SyncDirection))]
+	private ConveyorDirection CurrentDirection;
+
+	[SyncVar(hook = nameof(SyncStatus))]
+	private ConveyorStatus CurrentStatus;
+
+	private void SyncStatus(ConveyorStatus newStatus, ConveyorStatus oldStatus)
+	{
+		CurrentStatus = newStatus;
+		//do your thing
+		//all clients will be updated with this
+	}
+
+	private void SyncDirection(ConveyorDirection newDirection, ConveyorDirection oldDirection)
+	{
+		CurrentDirection = newDirection;
+		//do your thing
+		//all clients will be updated with this
+	}
+
+	[Server]
+	public void ServerChangeState(ConveyorStatus newStatus, ConveyorDirection newDirection)
+	{
+		CurrentStatus = newStatus;
+		CurrentDirection = newDirection;
+	}
+
+	[Server]
+	public void ServerChangeStatus(int newStatusTest)
+	{
+		Logger.Log("ServerChangeStatus: "+ newStatusTest);
+		if (newStatusTest == 0)
+		{
+			CurrentStatus = ConveyorStatus.Backward;
+		}
+		else if (newStatusTest == 1)
+		{
+			CurrentStatus = ConveyorStatus.Off;
+		}
+		else
+		{
+			CurrentStatus = ConveyorStatus.Forward;
+		}
+		Logger.Log("Current status " + CurrentStatus);
+	}
+
+	protected virtual void UpdateMe()
+	{
+		if (isServer)
+		{
+			timeElapsedServer += Time.deltaTime;
+			if (timeElapsedServer > AnimationSpeed)
+			{
+				//ChangeDirection();
+
+				//ServerChangeState(CurrentStatus, CurrentDirection);
+
+				ChangeAnimation();
+
+				DetectItems();
+
+				timeElapsedServer = 0;
+			}
+		}
+		else
+		{
+			timeElapsedClient += Time.deltaTime;
+			if (timeElapsedClient > AnimationSpeed)
+			{
+				ChangeAnimation();
+				timeElapsedClient = 0;
+			}
+		}
+	}
+
+	private void OnEnable()
+	{
+		UpdateManager.Add(CallbackType.UPDATE, UpdateMe);
+		OnStart();
+	}
+
+	void OnDisable()
+	{
+		UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+	}
+
+	private void OnStart()
+	{
+		registerTile = GetComponent<RegisterTile>();
+
+		CurrentDirection = MappedDirection;
+
+		//if (ConnectedSwitch == null) return;
+
+		//LastStatus = CurrentStatus;
+
+		//if (ConnectedSwitch.CurrentState == 0)
+		//{
+		//	CurrentStatus = ConveyorStatus.Backward;
+		//}
+		//else if (ConnectedSwitch.CurrentState == 1)
+		//{
+		//	CurrentStatus = ConveyorStatus.Off;
+		//}
+		//else
+		//{
+		//	CurrentStatus = ConveyorStatus.Forward;
+		//}
+
+		if (!isServer) return;
+
+		ServerChangeState(CurrentStatus, MappedDirection);
+	}
+
+	//void Start()
+	//{
+	//	registerTile = GetComponent<RegisterTile>();
+
+	//	CurrentDirection = MappedDirection;
+	//	CurrentStatus = MappedStatus;
+	//	LastDirection = CurrentDirection;
+	//	LastStatus = CurrentStatus;
+
+	//	if (!isServer) return;
+	//	ServerChangeState(MappedStatus, MappedDirection);
+	//}
+
+	private void ChangeDirection()
+	{
+		if (ConnectedSwitch == null) return;
+
+		LastStatus = CurrentStatus;
+
+		Logger.Log("Connected Switch: "+ ConnectedSwitch.CurrentState);
+
+		if (ConnectedSwitch.CurrentState == 0)
+		{
```

Comment:
Put this in the method:
![ChangeSwitch](https://user-images.githubusercontent.com/42844726/79424806-6c477d80-7fc9-11ea-9b5f-5bed4a896b52.PNG)